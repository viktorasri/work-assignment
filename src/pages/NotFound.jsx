import React from 'react';
import { Link } from 'react-router-dom';

import Page from '../components/Page';
import Button from '../components/Button';

const NotFound = () => {
  return (
    <Page title='404 Not Found'>
      <div className='row'>
        <div className='not-found block-center text-center'>
          <div className='not-found__sub-heading'>
            <h2>Error 404</h2>
          </div>
          <div className='not-found__heading'>
            <h1>Oops! Something went wrong.</h1>
          </div>

          <div className='d-flex justify-center'>
            <Link to='/'>
              <Button variant='primary'>Back to Homepage</Button>
            </Link>
          </div>
        </div>
      </div>
    </Page>
  );
};

export default NotFound;
