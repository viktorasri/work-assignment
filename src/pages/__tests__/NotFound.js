import React from 'react';
import { shallow } from 'enzyme';
import NotFound from '../NotFound';

describe('Page <NotFound />', () => {
  const wrapper = shallow(<NotFound />);

  it('expect to render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
