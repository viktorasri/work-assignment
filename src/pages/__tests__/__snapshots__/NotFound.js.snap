// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page <NotFound /> expect to render component 1`] = `
<Page
  description="Server explorer web app"
  title="404 Not Found"
>
  <div
    className="row"
  >
    <div
      className="not-found block-center text-center"
    >
      <div
        className="not-found__sub-heading"
      >
        <h2>
          Error 404
        </h2>
      </div>
      <div
        className="not-found__heading"
      >
        <h1>
          Oops! Something went wrong.
        </h1>
      </div>
      <div
        className="d-flex justify-center"
      >
        <Link
          to="/"
        >
          <Button
            variant="primary"
          >
            Back to Homepage
          </Button>
        </Link>
      </div>
    </div>
  </div>
</Page>
`;
